### PR TITLE
Tl/blocking-mode

### PR DESCRIPTION
**Pull request by tjlav5**
Update dependencies and enable experimental blocking mode

Dependency updates:
React@17
ReactDOM@17
Firebase@8
ReactFire@3

### CODE REVIEW

File: src/components/Firestore/DocumentEditor/types.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export type Store = {
 export function isPrimitive(
   data: FirestoreAny | DocumentPath
 ): data is PrimitiveValue {
-  return isNativePrimitive(data) || data instanceof DocumentPath;
+  return (isNativePrimitive(data) as any) || data instanceof DocumentPath;
```
Comment (yuchenshi) [inline src/components/Firestore/DocumentEditor/types.ts:49]: Is this really needed? It looks like `isNativePrimitive` has a return type of boolean, no?

**Review by yuchenshi — APPROVED** [review]

File: src/components/Firestore/Collection.tsx [hunk 2]
```diff
@@ -70,14 +72,17 @@ export function withCollectionState(
       // This works around an issue in React legacy mode that causes filters to
       // be ignored and listeners leaking.
       // TODO(yuchenshi): remove after switching to blocking / concurrent mode.
-      { startWithValue: { docs: [] } as any }
+      {
+        suspense: true,
+        initialData: { docs: [] } as any,
```
Comment (yuchenshi) [inline src/components/Firestore/Collection.tsx:77]: Please drop this and the comment block above. See the comments above for why.

Comment (tjlav5) [reply src/components/Firestore/Collection.tsx:77]: Done.

File: src/components/Firestore/DocumentEditor/types.ts (hunk 1 above)
Comment (tjlav5) [reply src/components/Firestore/DocumentEditor/types.ts:49]: weird - I removed it (which makes sense) - I'm not sure why the type-error came up before. Reverted.

**Pull request merged**